src/screens/ManutencaoIluminacao/ManutencaoItem.tsx

```diff
@@ -39,82 +42,143 @@ const materiais = [
-                                        <Text>Tensão: {material.tensao}</Text>
-                                        <Text>Base: {material.base}</Text>
-                                    </Body>
-                                </CardItem>
-                                <CardItem footer bordered>
-                                    <Item style={{ borderBottomColor: 'transparent' }}>
-                                        <Label>Qtde. Substituída:</Label>
-                                        <NumericInput
-                                            minValue={0}
-                                            editable={false}
-                                            rounded={true}
-                                            value={material.quantidade} 
-                                            onChange={quantidade => material.quantidade = quantidade} />
-                                    </Item>
-                                </CardItem>
-                            </Card>
-                        })
-                    }
-                    </ScrollView>
-                    <View style={{ flexDirection: 'row', marginVertical: 5 }}>
-                        <Button
-                            onPress={() => props.navigation.navigate('FotosItem')}
-                            style={{ flex: 1, marginRight: 3 }}
-                            iconLeft
-                            bordered
-                        >
-                            <Icon name="md-photos"/>
-                            <Text>Fotos</Text>
-                        </Button>
+    onChangeQuantidade = (idMaterial, quantidade) => {
+
+        const { materiais } = this.state;
+        const novosMateriais = materiais.map( material => {
+            if(material.id !== idMaterial) {
+                return material;
+            }
+            return {
+                ...material,
+                quantidade: quantidade
+            }
+        })
+        this.setState({materiais: novosMateriais})
+    }
+    
+    onPressBotaoOK = (idMaterial, quantidadeConfirmada) => {
+        const { materiais } = this.state;
+        const novosMateriais = materiais.map( material => {
+            if(material.id !== idMaterial) {
+                return material;
+            }
+            return {
+                ...material,
+                quantidadeConfirmada: !quantidadeConfirmada
+            }
+        })
+        this.setState({materiais: novosMateriais})
+    }
+    
+    render() {
+        
+        const {id} = this.props.navigation.state.params; 
+        
+         return (
+            <Container>
+                <HeaderNav title={"Manutenção Item #"+id} />
+                <Content padder>
+                    <KeyboardAvoidingView
+                        behavior="height"
+                    >
+                        <ScrollView>
+                        {
+                            this.state.materiais.map(material => {
```

Comment:
Não sei se é uma "prática recomendada oficialmente" mas eu tenho preferido deixar o "this.state" e o "this.props" sempre do lado de fora do _return_. Assim fica mais fácil caso você transforme em um componente _anexável_ e precise trocar de um pro outro.
Nesse caso eu colocaria antes do return uma declaração `const { materiais } = this.state` e usaria o `materiais` aqui no return